tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/action/start/StartActionTest.java

```diff
@@ -45,6 +45,7 @@ public void testStartActionNegative() {
         BAssertUtil.validateError(result, indx++, "action invocation as an expression not allowed here", 38, 32);
         BAssertUtil.validateError(result, indx++, "action invocation as an expression not allowed here", 39, 32);
         BAssertUtil.validateError(result, indx++, "action invocation as an expression not allowed here", 39, 37);
+        BAssertUtil.validateError(result, indx++, "variable assignment is required", 53, 9);
```

Comment:
This wouldn't get logged if we avoid logging an error if the type is semanticError right? I still think we should continue to not log the error if the type is semanticError. 

Even if we look at this example, we have 3 errors for the same line now, and they aren't quite independent, as in it is not guaranteed the user has to fix them separately.

Reply:
That again goes to my original point: If we do that, it will take the user two iterations to fix the code. In the first step, the user has to fix the semantic error in the expression itself, and then, has to fix the assignment required error. I think it's better to have all the possible errors in one pass, rather than having to fix them in multiple passes. Basically, for the given piece of code, all possible errors are given, considering each of the components in the line separately:
- `start` can only be used with a function/method/remote method call
- An action cannot be used with wait
- Wait is an expression and needs to be assigned

In this case I think the user does have to address all 3 to make it work. But I agree, there can be instances where the user doesn't have to fix each error individually to make it work. The point is, the user would have all the info at hand to make a decision on which path to take to fix the error.